pkg/state/appender.go

```diff
@@ -384,6 +384,82 @@ type appendTxParams struct {
 	initialisation   bool
 }
 
+func (a *txAppender) guessEthereumTransactionKind(ethTx *proto.EthereumTransaction, params *appendTxParams) (proto.EthereumTransactionKind, error) {
+	if ethTx.Data() == nil {
+		return proto.NewEthereumTransferWavesTx(), nil
+	}
+
+	selectorBytes := ethTx.Data()
+	selector, err := ethabi.NewSelectorFromBytes(selectorBytes[:4])
+	if err != nil {
+		return nil, errors.Errorf("failed to guess ethereum transaction kind, %v", err)
+	}
+
+	assetID := proto.AssetID{}
+	copy(assetID[:], ethTx.To()[:proto.AssetIDSize])
+	asset, err := a.state.AssetInfoByID(assetID, true)
+	if err != nil && !errors.Is(err, errs.UnknownAsset{}) {
+		return nil, errors.Errorf("failed to get asset info by ethereum recipient address %s, %v", ethTx.To().String(), err)
+	}
+
+	if ethabi.IsERC20Selector(selector) && !errors.Is(err, errs.UnknownAsset{}) {
```

Comment:
I have an mistake here. `!errors.Is(err, errs.UnknownAsset{})` is a wrong check. You should check that `err == nil`.

Reply:
Not really. "UnknownAsset" error is expected behavior. It should lead to the last case "Ethereum invoke script tx"

Reply:
If asset doesn't exist, error will be `errs.UnknownAsset{}`. But in this branch of code asset exists.